code/datums/manufacturing_requirements.dm

```diff
@@ -0,0 +1,303 @@
+	var/material_threshold = null
+
+	// ID must be defined, or else we have a problem
+	#ifdef CHECK_MORE_RUNTIMES
+	New()
+		. = ..()
+		if (isnull(id))
+			CRASH("[src] created with a null id")
+	#endif
+
+	proc/get_id()
+		return src.id
+
+	/// Returns whether or not the material in question matches our criteria. Defaults to true
+	proc/is_match(var/datum/material/M)
+		SHOULD_CALL_PARENT(TRUE)
+		if (isnull(M))
+			return FALSE
+		if (!isnull(src.material_flags) && !src.matches_flags(M.getMaterialFlags()))
+			return FALSE
+		return TRUE
+
+	/// Returns whether the material flags are matched. This will return true should any flag match.
+	proc/matches_flags(var/material_flags)
+		return material_flags & src.material_flags
+
+	/// Returns whether the material property matches the given criterion. Default behavior is to check if >=, override w/o calling parent for diff behavior.
+	proc/matches_property(var/datum/material/M)
+		return M.getProperty(src.material_property) >= src.material_threshold
+
+	any
+		name = "Any"
+		id = "any"
+
+/***************************************************************
+                      EXACT MATERIAL MATCH
+
+	     These are subtypes so only one instance is needed.
+       If you're adding one, just add the material id as the
+			       id property and type name.
+
+                    PLEASE ALPHABETIZE THANKS
+***************************************************************/
+	match_material
+		/// All you need to do is define the material id. we can take it from there ;P
+		New()
+			..()
+			var/datum/material/M = getMaterial(src.id)
+			src.name = M.getName()
+			src.material_id = initial(src.id)
+
+		is_match(var/datum/material/M)
+			if (!isnull(src.material_id) && !src.matches_id(M.getID()))
+				return FALSE
+			. = ..()
+
+		/// Returns whether the material id is an exact match for the required id.
+		proc/matches_id(var/material_id)
+			return src.material_id == material_id
+
+		bamboo
+			id = "bamboo"
+
+		blob
+			id = "blob"
+
+		butt
+			id = "butt"
+
+		cardboard
+			id = "cardboard"
+
+		cerenkite
+			id = "cerenkite"
+
+		char
+			id = "char"
+
+		claretine
+			id = "claretine"
+
+		cobryl
+			id = "cobryl"
+
+		ectoplasm
+			id = "ectoplasm"
+
+		electrum
+			id = "electrum"
+
+		erebite
+			id = "erebite"
+
+		exoweave
+			id = "exoweave"
+
+		gold
+			id = "gold"
+
+		honey
+			id = "honey"
+
+		ice
+			id = "ice"
+
+		iridiumalloy
+			id = "iridiumalloy"
+
+		koshmarite
+			id = "koshmarite"
+
+		miracle
+			id = "miracle"
+
+		molitz
+			id = "molitz"
+
+		neutronium
+			id = "neutronium"
+
+		pharosium
+			id = "pharosium"
+
+		plasmastone
+			id = "plasmastone"
+
+		plutonium
+			id = "plutonium"
+
+		starstone
+			id = "starstone"
+
+		syreline
+			id = "syreline"
+
+		telecrystal
+			id = "telecrystal"
+
+		uqill
+			id = "uqill"
+
+		viscerite
+			id = "viscerite"
```

Comment:
these should be autogenerated at runtime

Reply:
Might be for the best so that we don't need to special case having /datum/s and strings in the item_requirements of each manufacturer

Reply:
hold on i figured it out im dumb